flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -463,6 +475,22 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
                         new ResumeWrapper(controller.suspendDefaultAction(timer), timer)));
     }
 
+    protected void endData() throws Exception {
+        // Suspend the mailbox processor, it would be resumed in afterInvoke and finished
+        // after all records processed by the downstream tasks. We also suspend the default
```

Comment:
nitty nit:
`after all records processed by the downstream tasks` ->
`after all records are/will be processed by the downstream tasks`

Reply:
On a second thought. Are really disabling default action and suspending anything?

Reply:
I actually fixed that in the subsequent commit. I will remove it in this commit already, my bad.